### PR TITLE
fix(sdk-go): decode the annotations envelope and regenerate the REST client

### PR DESCRIPTION
**Pull request by 0xdeafcafe**
## What this does

The annotation endpoints return their payload inside a `data` envelope. The Go wrapper decoded the bare value, so `List`, `Get`, `ListByTrace`, `CreateForTrace` and `Update` did not match the response shape. `Delete` decodes nothing and was unaffected.

All five now go through one `decodeAnnotationEnvelope` helper, replacing five copies of the same anonymous struct.

**The envelope holds `data` as a pointer, and that is the point.** JSON decoding ignores keys it was not asked for, so a 2xx body in some other shape — an older self-hosted server still sending the bare value, or a future one that moves the payload again — decodes without error. Into a value that yields a zero-valued `Annotation` and a nil error, which is a worse failure than the one this PR set out to fix: the caller cannot tell it from a real annotation. Into a pointer it is a decode error naming the missing envelope.

Three tests hold that down, and all three fail if the nil check is removed.

`AnnotationParams` claimed every field was optional. The API requires both `comment` and `isThumbsUp` on create and update alike, so `Update`'s own documented example was a call the server always rejects with 400. Both are now documented as required, along with the `omitempty` consequence: an empty comment cannot be written.

## Regeneration

The committed REST client was eight spec commits behind the OpenAPI document. Regenerating drops 2 operations, adds 15, and picks up the `fields` and `evaluations` properties that scenarios and simulation runs gained. Those are additive.

## Breaking change

Five `Annotation` fields are values rather than pointers, because the schema marks them required: `Id`, `CreatedAt`, `UpdatedAt`, `ProjectId` and `TraceId`. `client.Annotation` is public surface, so callers that dereference them (`*a.Id`) should read the field directly. The nullable fields — `Comment`, `Email`, `IsThumbsUp`, `UserId` — are unchanged and stay pointers.

This PR touches `sdks/go` and nothing else, so that break stays inside one release component. `release-scope-guard` rejected an earlier revision that also carried `specs/` and `.github/` files. The spec and the CI job are stacked on this in #7990.

## Verification

- Every Go SDK module green under `./test-all.sh -race`, which is what CI runs.
- The envelope tests fail against the previous wrapper; the three missing-envelope tests fail if the nil guard is removed. Both checked by reverting, not assumed.
- `sdk-go-ci` green with `test`, `build` and `typecheck` executed rather than draft-skipped.

One `go-ci` run failed in `TestLLMProxy` under the Langy OTel relay, a package this branch does not touch. A re-run with no code change passed, so it is a flake there.

`golangci-lint` reports pre-existing issues in this module, including `bodyclose` false positives that map to the `decodeInto` call sites — the shared decoder closes the body in a deferred function the linter cannot follow. This module sits outside the `go-ci` lint scope, so none were enforced, and they are not in scope here.

## Known gap, not addressed here

The tests stub the wire shape as string literals, so they pin the SDK against a shape written by hand rather than against the OpenAPI document. A future schema change would update the generated client and leave these tests passing. Closing that means deriving fixtures from the schema, which is a larger change than this fix.

## Release note

The version this lands under needs a decision before release. `Release-As: 1.0.0` on #4998 is not being picked up, because that commit now sits 605 commits back and release-please's default commit search depth is 500, with no override configured. #6828 therefore proposes 0.4.0 from the one commit still inside the window, and four of the six Go SDK commits since 0.3.0 are absent from its changelog.

I have not touched the release configuration. Raising the search depth would also re-expose the two `Release-As` footers that the squash merge collapsed into that single commit.

### CODE REVIEW

File: sdks/go/client/services_test.go [hunk 1]
```diff
@@ -144,22 +180,51 @@ func TestAnnotations(t *testing.T) {
...
 
-		t.Run("when listing by trace", func(t *testing.T) {
+	t.Run("given a project with no annotations", func(t *testing.T) {
+		// @scenario "A project with no annotations yields an empty list"
+		t.Run("when listing annotations", func(t *testing.T) {
 			c := newTestClient(t, func(w http.ResponseWriter, r *http.Request) {
-				assert.Equal(t, "/api/annotations/trace/trace_1", r.URL.Path)
-				_, _ = w.Write([]byte(`[{"id":"ann_1"}]`))
+				_, _ = w.Write([]byte(`{"data":[]}`))
 			})
```
[the remark this answers is not in the document]
Comment (0xdeafcafe) [reply sdks/go/client/services_test.go:217]: Fixed. That handler now asserts the method and the path like its siblings.

While in there I also added the error-path coverage the suite was missing: a read that comes back without the envelope, a list in an unrelated shape, and a write answered with an empty body. Those were the cases that decoded silently into a zero-valued `Annotation` with a nil error, which is a worse outcome than the decode failure this PR set out to fix. `decodeAnnotationEnvelope` now holds `data` as a pointer and reports a decode error when it is absent, and removing that nil check makes all three new tests fail.
